wgengine/filter/filter_test.go

```diff
@@ -116,6 +123,67 @@ func TestFilter(t *testing.T) {
 	}
 }
 
+func TestNoAllocs(t *testing.T) {
+	acl := newFilter(t.Logf)
+
+	tcpPacket := tcpudp(TCP, 0x08010101, 0x01020304, 999, 22)
+	udpPacket := tcpudp(UDP, 0x08010101, 0x01020304, 999, 22)
+
+	tests := []struct {
+		name   string
+		in     bool
+		want   float64
+		packet []byte
+	}{
+		{"tcp_in", true, 0.0, tcpPacket},
+		{"tcp_out", false, 0.0, tcpPacket},
+		{"udp_in", true, 0.0, udpPacket},
+		// One alloc is inevitable (an lru cache update)
+		{"udp_out", false, 1.0, udpPacket},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := testing.AllocsPerRun(1000, func() {
+				var q QDecode
+				if test.in {
+					acl.RunIn(test.packet, &q, 0)
+				} else {
+					acl.RunOut(test.packet, &q, 0)
+				}
+			})
+
+			// It is OK to allocate once per 1000 runs or so (rate-limited logging).
+			// We also bound got in both directions here to ensure we understand
+			// our allocation behavior: if it improves, we should change the excepted values.
+			if got-test.want > 0.001 || test.want-got > 0.001 {
```

Comment:
Just use `!=` or `> (maybe the text `want` value should be the upper bound of allocations?).  It's only of type float for historical reasons. It's actually an integer. ("Although the return value has type float64, it will always be an integral value.", say the docs)

Reply:
The `!=` was so that we notice if we manage to, e.g. remove the one remaining allocation. At the same time, I guess the person doing it would be interested in benchmarking the outcome anyway, so I changed it to `>`.